test/abilities/neutralizing-gas.test.ts

```diff
@@ -0,0 +1,197 @@
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override
+      .moveset([MoveId.SPLASH])
+      .ability(AbilityId.NEUTRALIZING_GAS)
+      .battleType("single")
+      .disableCrits()
+      .enemySpecies(SpeciesId.MAGIKARP)
+      .enemyAbility(AbilityId.BALL_FETCH)
+      .enemyMoveset(MoveId.SPLASH);
+  });
+
+  it("should prevent other abilities from activating", async () => {
+    game.override.enemyAbility(AbilityId.INTIMIDATE);
+    await game.classicMode.startBattle(SpeciesId.FEEBAS);
+
+    game.move.select(MoveId.SPLASH);
+    await game.toEndOfTurn();
+
+    // Intimidate is suppressed, so the attack stat should not be lowered
+    expect(game.field.getPlayerPokemon().getStatStage(Stat.ATK)).toBe(0);
+  });
+
+  it("should allow the user's passive to activate", async () => {
+    game.override.passiveAbility(AbilityId.INTREPID_SWORD);
+    await game.classicMode.startBattle(SpeciesId.FEEBAS);
+
+    game.move.select(MoveId.SPLASH);
+    await game.toEndOfTurn();
+
+    expect(game.field.getPlayerPokemon().getStatStage(Stat.ATK)).toBe(1);
+  });
+
+  // TODO: https://github.com/Despair-Games/poketernity/issues/1211
+  it.todo("should activate before other abilities", async () => {
+    game.override.enemySpecies(SpeciesId.ACCELGOR).enemyLevel(100).enemyAbility(AbilityId.INTIMIDATE);
+
+    await game.classicMode.startBattle(SpeciesId.FEEBAS);
+
+    game.move.select(MoveId.SPLASH);
+    await game.toEndOfTurn();
+
+    // Intimidate is suppressed even when the user's speed is lower
+    expect(game.field.getPlayerPokemon().getStatStage(Stat.ATK)).toBe(0);
+  });
+
+  it("should activate other abilities when removed", async () => {
+    game.override
+      .enemyAbility(AbilityId.INTREPID_SWORD)
+      .enemyPassiveAbility(AbilityId.DAUNTLESS_SHIELD)
+      .enemyMoveset(MoveId.ENTRAINMENT);
+
+    await game.classicMode.startBattle(SpeciesId.FEEBAS);
+
+    const enemyPokemon = game.field.getEnemyPokemon();
+    expect(enemyPokemon.getStatStage(Stat.ATK)).toBe(0);
+    expect(enemyPokemon.getStatStage(Stat.DEF)).toBe(0);
+
+    game.move.select(MoveId.SPLASH);
+    await game.toEndOfTurn();
+    // Enemy removes user's ability, so both abilities are activated
+    expect(enemyPokemon.getStatStage(Stat.ATK)).toBe(1);
+    expect(enemyPokemon.getStatStage(Stat.DEF)).toBe(1);
+  });
+
+  it("should not activate the user's other ability when removed", async () => {
+    game.override.passiveAbility(AbilityId.INTIMIDATE).enemyMoveset(MoveId.ENTRAINMENT);
+
+    await game.classicMode.startBattle(SpeciesId.FEEBAS);
+    // Neutralising gas user's passive is still active
+    const enemyPokemon = game.field.getEnemyPokemon();
+    expect(enemyPokemon.getStatStage(Stat.ATK)).toBe(-1);
```

Comment:
There is a `toHaveStatStage` custom matcher that improves readability 

```suggestion
    expect(enemyPokemon).toHaveStatStage(Stat.ATK, -1);
```